spec/models/service_template_transformation_plan_request_spec.rb

```diff
@@ -0,0 +1,39 @@
+describe ServiceTemplateTransformationPlanRequest do
+  let(:vms) { Array.new(3) { FactoryGirl.create(:vm_or_template) } }
+  let(:vm_requests) do
+    %w(Queued Failed Approved).zip(vms).collect do |status, vm|
+      ServiceResource.new(:resource => vm, :status => status)
+    end
+  end
+  let(:plan) { FactoryGirl.create(:service_template_transformation_plan, :service_resources => vm_requests) }
+  let(:request) { FactoryGirl.create(:service_template_transformation_plan_request, :source => plan) }
+
+  describe '#requested_task_idx' do
+    it 'selects approved vm requests' do
+      expect(request.requested_task_idx.first).to have_attributes(:resource => vms[2], :status => 'Approved')
+    end
+  end
+
+  describe 'customize_request_task_attributes' do
+    it 'overrides #customize_request_task_attributes' do
+      expect(described_class.instance_methods(false)).to include(:customize_request_task_attributes)
```

Comment:
We should be testing the result of the method and not just the existence of the override.